linux/Platform.c

```diff
@@ -620,124 +620,102 @@ bool Platform_getNetworkIO(NetworkIOData* data) {
 // READ FROM /proc
 // ----------------------------------------
 
-static unsigned long int parseBatInfo(const char* fileName, const unsigned short int lineNum, const unsigned short int wordNum) {
-   const char batteryPath[] = PROC_BATTERY_DIR;
-   DIR* batteryDir = opendir(batteryPath);
-   if (!batteryDir)
-      return 0;
-
-   char* batteries[MAX_BATTERIES];
-   unsigned int nBatteries = 0;
-   memset(batteries, 0, MAX_BATTERIES * sizeof(char*));
-
-   while (nBatteries < MAX_BATTERIES) {
-      const struct dirent* dirEntry = readdir(batteryDir);
-      if (!dirEntry)
-         break;
-
-      const char* entryName = dirEntry->d_name;
-      if (!String_startsWith(entryName, "BAT"))
-         continue;
-
-      batteries[nBatteries] = xStrdup(entryName);
-      nBatteries++;
-   }
-   closedir(batteryDir);
-
-   unsigned long int total = 0;
-   for (unsigned int i = 0; i < nBatteries; i++) {
-      char infoPath[30];
-      xSnprintf(infoPath, sizeof infoPath, "%s%s/%s", batteryPath, batteries[i], fileName);
-
-      FILE* file = fopen(infoPath, "r");
-      if (!file)
-         break;
-
-      char* line = NULL;
-      for (unsigned short int j = 0; j < lineNum; j++) {
-         free(line);
-         line = String_readLine(file);
-         if (!line)
+static double Platform_Battery_getProcBatInfo(void) {
+    unsigned long int   totalFull = 0, totalRemain = 0;
+    char                filePath[256] = {0}, *buf = NULL, buffer[1024] = {0}, field[100] = {0};
+    const char          batteryPath[] = PROC_BATTERY_DIR;
+    DIR*                batteryDir = opendir(batteryPath);
+    int                 val = 0;
+    ssize_t             r = 0;
+    if (!batteryDir)
+        return 0;
+    for(;;){
```

Comment:
Use `do { /* ... */ } while(direntry);` tail-controlled loop.
Break early if reading `dirEntry` in the next line returns `NULL`.